extensions/walletProviders/tron/src/extension.ts

```diff
@@ -0,0 +1,223 @@
+   * @returns {string} An Ethereum address in string format.
+   */
+  public async generateAddressFromMnemonic(mnemonic: string, index: number, path?: string) {
+    path = path || TRON_DERIVATION_PATH
+
+    const seed = await mnemonicToSeed(mnemonic)
+
+    const xpub = BIP32Factory(ecc)
+        .fromSeed(seed)
+        .derivePath(path)
+        .neutered()
+        .toBase58()
+
+    const pub = BIP32Factory(ecc)
+        .fromBase58(xpub)
+        .derive(index)
+        .publicKey
+
+    return TronWeb.address.fromHex(generateAddress(pub))
+  }
+
+  /**
+   * Generates an address from an extended public key (xpub) and an index.
+   * @param {string} xpub - The extended public key.
+   * @param {number} index - The index to derive the address from.
+   * @returns {string} An Ethereum address in string format.
+   */
+  public generateAddressFromXpub(xpub: string, index: number) {
+    let w: BIP32Interface
+    if (xpub.length === 130 && isHex(xpub)) {
+      w = BIP32Factory(ecc).fromPublicKey(Buffer.from(xpub.slice(0, 66), 'hex'), Buffer.from(xpub.slice(-64), 'hex'))
+    } else if (xpub.length === 111 && isBase58(xpub)) {
+      w = BIP32Factory(ecc).fromBase58(xpub)
+    } else {
+      throw new Error('Unknown xpub format')
+    }
+    return TronWeb.address.fromHex(generateAddress(w.derive(index).publicKey))
+  }
+
+  /**
+   * Generates an address from a given private key.
+   * @param {string} privateKey - The private key in string format.
+   * @returns {string} An Ethereum address in string format.
+   */
+  public generateAddressFromPrivateKey(privateKey: string) {
+    return TronWeb.address.fromPrivateKey(privateKey)
+  }
+
+  /**
+   * Generates an EVM-compatible wallet, which includes an address, private key, and a mnemonic.
+   * @returns {TronWallet} An object containing address, private key, and mnemonic.
+   */
+  public async getWallet() {
+    const mnemonic = this.generateMnemonic()
+    const privateKey = await this.generatePrivateKeyFromMnemonic(mnemonic, 0)
+    const address = this.generateAddressFromPrivateKey(privateKey)
+
+    return { address, privateKey, mnemonic } as TronWallet
+  }
+
+  /**
+   * Signs and broadcasts a Tron transaction payload.
+   * @param {TronTxPayload} payload - The Tron transaction payload, which includes private key and transaction details.
+   * @returns {Promise<string>} A promise that resolves to the transaction hash.
+   */
+  public async signAndBroadcast(payload: TronTxPayload): Promise<string> {
+    const value = this.getTransactionValue(payload)
+    const metaData = await this.getHeaderInfo()
+    const tx = this.getTx(value, metaData)
+
+    const signedTransaction = TronWeb.utils.crypto.signTransaction(payload.fromPrivateKey, tx)
+
+    const response = await this.tronRpc.broadcastTransaction(signedTransaction)
+
+    if(!response?.txid){
+        throw new Error(JSON.stringify(response))
+    }
+
+    return response?.txid
```

Comment:
No need for question mark since you're already throwing error if it doesn't exist.